### PR TITLE
feat: Problem 508 GDoc Parser Refactor - Paragraph

### LINKED ISSUE #508: Problem: GDoc parser is not well structured - Proper Type Assertion
# Problem statement
GDoc parser is lack of proper type assertion, and it lacks some best practices.

# Proposed solution
Add proper types to the parsed elements

### ETA
Expected time to finish: 2 hr
Time taken so far: 2 hrs

### PR DESCRIPTION
**Pull request by accompany1205**
Resolve #508 

ETA:
Expected time to finish: 3 hr
Time taken so far: 2 hrs

- [x] Refactor paragraph parser.

### CODE REVIEW

File: src/routes/api/google-conversion/+server.ts [hunk 1]
```diff
@@ -397,115 +459,60 @@ function parseParagraph(
...
-              .join(' ')
-              .replace(' .', '.')
-              .replace(' ,', ',')
-              .slice(2),
-            caption: '',
+            text: trimJoinArray(
+              quoteElements
+                .map((element) => element[tag])
+                .filter((element) => Boolean(element.length))
+            ).slice(2),
```
Comment (georgeciubotaru) [inline src/routes/api/google-conversion/+server.ts:475]: missing `caption` field

File: src/routes/api/google-conversion/+server.ts [hunk 2]
```diff
@@ -397,115 +459,60 @@ function parseParagraph(
...
           type: 'quote',
           data: {
-            text: tagContent
-              .map((el) => el[tag])
-              .filter((el) => el.length > 0)
-              .join(' ')
-              .replace(' .', '.')
-              .replace(' ,', ','),
-            caption: '',
+            text: trimJoinArray(
```
Comment (georgeciubotaru) [inline src/routes/api/google-conversion/+server.ts:505]: same here

**Comment by georgeciubotaru** [comment]
@accompany1205 Feel free to check for the output structure by using the actual API endpoint and passing the JSON object, then use `https://www.jsondiff.com/` to compare them

**Comment by accompany1205** [comment]
> @accompany1205 Feel free to check for the output structure by using the actual API endpoint and passing the JSON object, then use `https://www.jsondiff.com/` to compare them

All the captions are empty string. Why do we add those to the parsed JSON?

**Review by georgeciubotaru — APPROVED** [review]

**Pull request merged**